web/src/__tests__/date-utils.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { DateFormat, dateToFormat } from '../date-utils';
+
+const mockDate = new Date('2023-02-22T08:38:36Z');
+const mockDateWithMillis = new Date('2023-02-22T08:38:36.001Z');
+const invalidDate = new Date('2023-02-22T24:38:36.001Z');
+
+describe('date utils', () => {
+  it('should format a date correctly', () => {
+    [
+      { format: DateFormat.DateMed, output: '2023-02-22' },
+      { format: DateFormat.Full, output: 'Feb 22, 2023, 08:38:36.000' },
+      { format: DateFormat.TimeFull, output: '08:38:36.000' },
+      { format: DateFormat.TimeMed, output: '08:38:36' },
+      { format: DateFormat.TimeShort, output: '08:38' },
+    ].forEach(({ format, output }) => {
+      expect(dateToFormat(mockDate, format)).toEqual(output);
+    });
+  });
+
+  it('should format a date with milliseconds correctly', () => {
```

Comment:
It would be good to add a couple of cases where the minutes / seconds are less than 10.